### PR TITLE
refactor and add url generator

### PR DESCRIPTION
**Pull request by kimjngyun**
Added typescript and react-query
Added the URLGenerator component so that user can edit URLs that contain variables.

https://github.com/user-attachments/assets/cb3cb925-97a9-49e7-9f57-12c97f675b91

### CODE REVIEW

**Review by lucetre — APPROVED** [review]
LGTM

File: src/components/URLGenerator.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+const extractVariables = (template: string) => {
+  const pattern = /\{([^}]+)\}/g;
+  const extractedVars = [];
+  let match;
+  while ((match = pattern.exec(template)) !== null) {
+    extractedVars.push(match[1]);
+  }
+  return extractedVars;
+};
```
Comment (lucetrez) [inline src/components/URLGenerator.tsx:11]: 💯

File: src/lib/utils.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    if (match) {
+      console.log(`Matched pattern: ${pattern}`);
+      console.log(`Redirect URL before replacement: ${url}`);
+      // Replace '{$1}' with the captured group
+      return url.replace("{$1}", match[1]);
+    }
+  }
+  return null;
+};
```
Comment (kimjngyun) [inline src/lib/utils.ts:21]: need to update?
this function covers only $1 case.

Comment (lucetrez) [reply src/lib/utils.ts:21]: Yes we need to cover $2...$N, but less priority atm.

**Pull request merged**